wire-ios-sync-engine/Source/Use cases/ConversationFolderSelectionUseCase.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireDataModel
+
+// MARK: - ConversationFolderSelectionUseCaseProtocol
+
+public protocol ConversationFolderSelectionUseCaseProtocol {
```

Comment:
I don't think this protocol is needed

Instead I expect you will need to define a protocol in your `WireUI` target and conform the protocol to it outside (i.e Wire-iOS target).

Personally I would avoid creating the protocol until you actually need it.

Reply:
Let's keep it for now and see later if we indeed need it or not. Unclear for me still.